sqld/src/replication/primary/logger.rs

```diff
@@ -790,6 +874,29 @@ impl ReplicationLogger {
     }
 }
 
+fn checkpoint_db(data_path: &Path) -> anyhow::Result<()> {
+    unsafe {
+        let conn = rusqlite::Connection::open(data_path)?;
+        conn.execute("VACUUM", ())?;
```

Comment:
Any reason for calling `VACUUM` before checkpointing rather than after? After would have better guarantees for actually minimizing the database file, since checkpointing itself may create lots of writes to the main db file. A prime example that comes to mind is calling checkpoint_db() on a database with an empty main file and 1GiB of WAL that waits to be checkpointed

Reply:
oh and btw -- imo, `VACUUM` should be called in a best-effort manner, ignoring the error. If for some reason it returned a `SQLITE_BUSY` or whatever, it's still a good idea to try checkpointing